### PR TITLE
Introduce swap background data augmentation.

### PR DESCRIPTION
**Pull request by lengzq**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by lengzq** [comment]
@tanzhenyu

File: keras_cv/layers/preprocessing3d/swap_background_test.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        outputs = add_layer(inputs)
+        # The following points in additional_point_clouds.
+        # [0, 2, 1, 3, 4], -> kept because it is in additional_point_clouds [0, 0, 1, 4, 4, 4, 0, 1].
+        # [0, 0, 2, 0, 2] -> removed because it is a background point (not in any bounding_boxes and additional_point_clouds).
+        # [0, 11, 2, 3, 4] -> removed because it is a background point (not in any bounding_boxes and additional_point_clouds).
+        # [100, 101, 2, 3, 4] -> kept because it is in additional_point_clouds [100, 100, 2, 5, 5, 5, 0, 1].
+        # [10, 10, 10, 10, 10] -> removed because it is a background point (not in any bounding_boxes and additional_point_clouds).
+        # The following points in point_clouds.
+        # [0, 1, 2, 3, 4] -> removed because it is in bounding_boxes [0, 0, 0, 4, 4, 4, 0, 1].
+        # [10, 1, 2, 3, 4] -> kept because it is a background point (not in any bounding_boxes and additional_point_clouds).
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing3d/swap_background_test.py:96]: +100 on very detailed comments!!

File: keras_cv/layers/preprocessing3d/base_augmentation_layer_3d.py [hunk 2]
```diff
@@ -18,7 +18,10 @@
 BOUNDING_BOXES = "bounding_boxes"
 OBJECT_POINT_CLOUDS = "object_point_clouds"
 OBJECT_BOUNDING_BOXES = "object_bounding_boxes"
+ADDITIONAL_POINT_CLOUDS = "additional_point_clouds"
+ADDITIONAL_BOUNDING_BOXES = "additional_bounding_boxes"
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing3d/base_augmentation_layer_3d.py:22]: Isn't additional point clouds the same as object point clouds? I was under the impression that group_points_by_boxes in only called for the additional point clouds

Comment (lengzq) [reply keras_cv/layers/preprocessing3d/base_augmentation_layer_3d.py:22]: Yes, OBJECT_POINT_CLOUDS are extracted from ADDITIONAL_POINT_CLOUDS.

all_point_clouds = [point_clouds, additional_point_cloud]

We use group_by function to extract objects from additional_point_cloud for copy-paste data augmentation.

File: keras_cv/layers/preprocessing3d/swap_background.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            transformation[ADDITIONAL_BOUNDING_BOXES]
+        )
+        return (
+            point_clouds.to_tensor(shape=original_point_clouds_shape),
+            bounding_boxes.to_tensor(shape=original_bounding_boxes_shape),
+        )
+
+    def _augment(self, inputs):
+        result = inputs
+        point_clouds = inputs.get(POINT_CLOUDS, None)
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing3d/swap_background.py:148]: Is there a reason we can't just use bracket indexing here? e.g. inputs[POINT_CLOUDS]

File: keras_cv/layers/preprocessing3d/base_augmentation_layer_3d.py (hunk 2 above)
Comment (ianstenbit) [reply keras_cv/layers/preprocessing3d/base_augmentation_layer_3d.py:22]: I think given the complexity of the relationship between e.g. POINT_CLOUDS, OBJECT_POINT_CLOUDS, and ADDITIONAL_POINT_CLOUDS, we should add some documentation in the base layer about what they mean and how to use them.

(In another PR is fine with me)

**Review by ianstenbit — APPROVED** [review]
Thank you!

File: keras_cv/layers/preprocessing3d/base_augmentation_layer_3d.py (hunk 2 above)
Comment (lengzq) [reply keras_cv/layers/preprocessing3d/base_augmentation_layer_3d.py:22]: Agree. I will document those in a different PR once we check in all the augmentation layers.

File: keras_cv/layers/preprocessing3d/swap_background.py (hunk 3 above)
Comment (lengzq) [reply keras_cv/layers/preprocessing3d/swap_background.py:148]: Updated

**Pull request merged**